src/datasets/streaming.py

```diff
@@ -112,3 +112,9 @@ def extend_dataset_builder_for_streaming(builder: "DatasetBuilder"):
                 internal_import_name = imports[1]
                 internal_module_name = ".".join(builder.__module__.split(".")[:-1] + [internal_import_name])
                 extend_module_for_streaming(internal_module_name, use_auth_token=builder.use_auth_token)
+
+    # we need to patch autofolder.py manually because it's a parent of Image and Audio folder and it
+    # contains generation of examples (which uses `os` functions)
+    autofolder_module = [cls.__module__ for cls in type(builder).__mro__ if cls.__name__ == "AutoFolder"]
```

Comment:
Just in case someone subclasses a class named "AutoFolder" that is not the one in `datasets`:
```suggestion
    # contains generation of examples (which uses `os` functions)
    from .packaged_modules.base.autofolder import AutoFolder

    autofolder_module = [cls.__module__ for cls in type(builder).__mro__ if cls is AutoFolder
```